controllers/windowsmachine_controller.go

```diff
@@ -515,3 +508,21 @@ func (r *WindowsMachineReconciler) isWindowsMachineHealthy(machine *mapi.Machine
 
 	return true
 }
+
+// getInternalIPAddress returns the internal IP address of the Machine
+func getInternalIPAddress(addresses []core.NodeAddress) (string, error) {
+	// Get the IP address associated with the Windows machine, if not error out to requeue again
+	if len(addresses) == 0 {
+		return "", errors.New("no IP addresses defined")
+	}
+	ipAddress := ""
+	for _, address := range addresses {
+		if address.Type == core.NodeInternalIP {
+			ipAddress = address.Address
+		}
```

Comment:
```suggestion
		if address.Type == core.NodeInternalIP || address.Type == core.NodeHostName {
			ipAddress = address.Address
		}
```
@aravindhp don't we support the NodeHostName address type here too? (If I'm not mistaken some platforms set the InternalP, while others set HostName)

Reply:
If we do, why is this not present in the original code? I am bit hesitant to change behavior in this PR.

Reply:
If the case never occurs where ONLY hostname (and no InternalIP) is set, this can be disregarded.
@jcpowermac can you confirm this never happens on vSphere?

Reply:
And if it does happen then we should introduce checking for hostname in a separate PR as it is a clear change in WMCO behavior.